app/features/wallet/wallet.tsx

```diff
@@ -71,14 +109,27 @@ const Wallet = ({ children }: PropsWithChildren) => {
 
   useSyncThemeWithDefaultCurrency();
 
-  useTrackAccounts();
-  useTrackPendingCashuReceiveQuotes();
-  useTrackPendingCashuTokenSwaps();
-  useTrackUnresolvedCashuSendQuotes();
-  useTrackUnresolvedCashuSendSwaps();
-
   const isLead = useTakeTaskProcessingLead();
 
+  const accountsSubscription = useTrackAccounts();
+  const pendingCashuReceiveQuotesSubscription =
+    useTrackPendingCashuReceiveQuotes();
+  const pendingCashuTokenSwapsSubscription = useTrackPendingCashuTokenSwaps();
+  const unresolvedCashuSendQuotesSubscription =
+    useTrackUnresolvedCashuSendQuotes();
+  const unresolvedCashuSendSwapsSubscription =
+    useTrackUnresolvedCashuSendSwaps();
+
+  if (
```

Comment:
this approach seems simpler but tbh I don't like this part nor the fact that we have one more place where we conditionally block app rendering and display loading screen. it takes us one step further from having less loading screens shown